openstacksdk_plugin/utils.py

```diff
@@ -148,3 +158,52 @@ def update_runtime_properties(keys=None):
     keys = keys or {}
     for key, value in keys.items():
         ctx.instance.runtime_properties[key] = value
+
+
+def add_resource_list_to_runtime_properties(openstack_type_name, object_list):
+    objects = []
+    for obj in object_list:
+        if type(obj) not in [str, dict]:
+            obj = obj.to_dict()
+        objects.append(obj)
+
+    key_list = '{0}_list'.format(openstack_type_name)
+    ctx.instance.runtime_properties[key_list] = objects
+
+
+def validate_resource(resource, openstack_type):
+    ctx.logger.debug(
+        'validating resource {0} (node {1})'
+        ''.format(openstack_type, ctx.node.id)
+    )
+    openstack_type_plural = resource.resource_plural(openstack_type)
+
+    resource_list = list(resource.list())
+
+    # This is the available quota for provisioning the resource
+    resource_amount = len(resource_list)
```

Comment:
What type is `resource.list()` and how does `len` handle it?

Reply:
resource.list will return generator, so we need to evaluate that and get the number of instances there so that we can check them